### PR TITLE
[AppConfig] fixes live test failure

### PR DESCRIPTION
**Pull request by mssfang**
Fixes the tests failures on this PR:
https://github.com/Azure/azure-sdk-for-java/pull/17271

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java [hunk 1]
```diff
@@ -539,15 +542,36 @@ public void listWithMultipleKeys(HttpClient httpClient, ConfigurationServiceVers
...
+        String key = getKey();
+        String key2 = getKey();
+
+        // Delete all existing settings in the resource
+        final PagedFlux<ConfigurationSetting> configurationSettingPagedFlux2 = client.listConfigurationSettings(null);
+        for (ConfigurationSetting setting : configurationSettingPagedFlux2.toIterable()) {
+            StepVerifier.create(client.deleteConfigurationSettingWithResponse(setting, false))
+                .assertNext(response -> assertConfigurationEquals(setting, response))
+                .verifyComplete();
+        }
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java:556]: If we wanted this to be fully reactive:

```suggestion
Mono<Void> deleteAllSettingsMono = client.listConfigurationSettings(null)
    .flatMap(setting -> client.deleteConfigurationSettingWithResponse(setting, false))
    .then();
    
StepVerifier.create(deleteAllSettingsMono)
   .verifyComplete();
```

This code doesn't need to worry about asserting values returned by the delete.

**Review by alzimmermsft — APPROVED** [review]

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java [hunk 2]
```diff
@@ -539,15 +542,36 @@ public void listWithMultipleKeys(HttpClient httpClient, ConfigurationServiceVers
 
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.data.appconfiguration.TestHelper#getTestParameters")
-    public void listConfigurationSettings(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
+    public void listConfigurationSettingsWithNullSelector(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
         client = getConfigurationAsyncClient(httpClient, serviceVersion);
+        String key = getKey();
+        String key2 = getKey();
```
Comment (hemanttanwar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java:548]: could these variable be final ?

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java [hunk 3]
```diff
@@ -539,15 +542,36 @@ public void listWithMultipleKeys(HttpClient httpClient, ConfigurationServiceVers
...
 
-        SettingSelector settingSelector = new SettingSelector();
-        settingSelector.setKeyFilter("prodDBConnection");
-        com.azure.core.util.Context context = new com.azure.core.util.Context(SettingFields.KEY, SettingFields.VALUE);
+        // Delete all existing settings in the resource
+Mono<Void> deleteAllSettingsMono = client.listConfigurationSettings(null)
```
Comment (hemanttanwar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java:551]: nit : `Mono<Void>` is not starting from connect indentation. Is it just git showing this way or this should be fixed ?

**Review by hemanttanwar — APPROVED** [review]

**Pull request merged**